pkg/controllers/tagging/tagging_controller_test.go

```diff
@@ -135,15 +135,26 @@ func Test_NodesJoiningAndLeaving(t *testing.T) {
 			} else {
 				tc.enqueueNode(testcase.currNode, tc.untagNodeResources)
 			}
-			tc.process()
+
+			for tc.workqueue.Len() > 0 {
+				tc.process()
+
+				// sleep briefly because of exponential backoff when requeueing failed workitem
+				// resulting in workqueue to be empty if checked immediately
+				time.Sleep(1500 * time.Millisecond)
+			}
```

Comment:
Can you please explain why this is needed?

Reply:
It is because of [this](https://github.com/kubernetes/cloud-provider-aws/blob/master/pkg/controllers/tagging/tagging_controller.go#L145). [AddRateLimited](https://github.com/kubernetes/kubernetes/blob/a88462e901710ab5df1fa6e99a046aa921dcdd55/staging/src/k8s.io/client-go/util/workqueue/rate_limiting_queue.go#L24) only adds back the item to the queue after some period of wait time, and it gets longer each try. As such, I put this sleep function in here so that we are sure to capture all retries for testing. I admit this depends on the implementation of the workqueue, which is not ideal because the test could fail not for its own reason.